code/__HELPERS/mobs.dm

```diff
@@ -230,8 +230,12 @@ GLOBAL_LIST_EMPTY(species_list)
 		delay *= user.cached_multiplicative_actions_slowdown
 
 	var/datum/progressbar/progbar
+	var/datum/cogbar/cog
+
 	if(progress)
 		progbar = new(user, delay, target || user)
+		if(visibility == DO_PUBLIC && delay >= 1 SECONDS)
+			cog = new(user)
```

Comment:
Seems like you've located this code such that if we merge your _other_ PR then if NPC is performing a do_after, it won't display the cog. 
This seems undesirable to me, surely if we want an action to be visible we want it to be visible if it is being performed autonomously too?

Reply:
That's what the "hidden" flag is for, no? 
If it should be invisible when an NPC is doing it why would we want it to be visible if a player-controlled one is doing it?

Reply:
good point. with the other PR, i will format it to only not create the progbar.